schema/encoding/jsonschema/jsonschema.go

```diff
@@ -18,9 +18,34 @@ var (
 	ErrNotImplemented = errors.New("not implemented")
 )
 
+// Wrap IO writer so we can consolidate error handling
+// in a single place. Also track properties written
+// so we know when to emit a separator.
 type errWriter struct {
-	w   io.Writer
-	err error
+	w          io.Writer // writer instance
+	err        error     // track errors
+	properties int       // track properties written
+}
+
+// Comma optionally outputs a comma.
+// Invoke this when you're about to write a property.
+// Tracks how many have been written and emits if not the first.
+func (ew *errWriter) Comma() {
```

Comment:
Although it does not matter to much as the errWriter type is private, shoul this method still be private for consistency with writeFormat?

For a public interface, I would use a bit more time on defining a better interface, as can be discussed further in #35 and #51.